core/crypto/nike/hybrid/hybrid.go

```diff
@@ -0,0 +1,210 @@
+//go:build ctidh
+// +build ctidh
+
+package hybrid
+
+import (
+	"encoding/base64"
+
+	"github.com/katzenpost/katzenpost/core/crypto/nike"
+	"github.com/katzenpost/katzenpost/core/crypto/nike/ctidh"
+	"github.com/katzenpost/katzenpost/core/crypto/nike/ecdh"
+	"github.com/katzenpost/katzenpost/core/crypto/rand"
+)
+
+var CTIDHX25519 nike.Scheme = &scheme{
+	name:   "CTIDH-X25519",
+	first:  ctidh.CTIDHScheme,
+	second: ecdh.NewEcdhNike(rand.Reader),
+}
+
+type publicKey struct {
+	scheme *scheme
+	first  nike.PublicKey
+	second nike.PublicKey
+}
+
+type privateKey struct {
+	scheme *scheme
+	first  nike.PrivateKey
+	second nike.PrivateKey
+}
+
+type scheme struct {
+	name   string
+	first  nike.Scheme
+	second nike.Scheme
+}
+
+func (s *scheme) Name() string {
+	return s.name
+}
+
+func (s *scheme) PublicKeySize() int {
+	return s.first.PublicKeySize() + s.second.PublicKeySize()
+}
+
+func (s *scheme) PrivateKeySize() int {
+	return s.first.PrivateKeySize() + s.second.PrivateKeySize()
+}
+
+func (s *scheme) NewKeypair() (nike.PrivateKey, nike.PublicKey) {
+	privKey1, pubKey1 := s.first.NewKeypair()
+	privKey2, pubKey2 := s.second.NewKeypair()
+	return &privateKey{
+			scheme: s,
+			first:  privKey1,
+			second: privKey2,
+		}, &publicKey{
+			scheme: s,
+			first:  pubKey1,
+			second: pubKey2,
+		}
+}
+
+func (s *scheme) DeriveSecret(privKey nike.PrivateKey, pubKey nike.PublicKey) []byte {
+	return append(privKey.(*privateKey).scheme.first.DeriveSecret(privKey.(*privateKey).first, pubKey.(*publicKey).first),
+		privKey.(*privateKey).scheme.second.DeriveSecret(privKey.(*privateKey).second, pubKey.(*publicKey).second)...)
+}
+
+func (s *scheme) DerivePublicKey(privKey nike.PrivateKey) nike.PublicKey {
+	return &publicKey{
+		scheme: s,
+		first:  privKey.(*privateKey).scheme.first.DerivePublicKey(privKey.(*privateKey).first),
+		second: privKey.(*privateKey).scheme.second.DerivePublicKey(privKey.(*privateKey).second),
+	}
+}
+
+func (s *scheme) Blind(groupMember []byte, blindingFactor []byte) (blindedGroupMember []byte) {
```

Comment:
that `BlindingFactor` takes `[s.PrivateKeySize]byte` seems like an implementation detail, it feels like there should be a constant/method for that in the interface.
This `Blind` takes a `[]byte`, and assumes that a `scheme` uses the same code to blind both `nike.PublicKey` and `nike.PrivateKey`, which also seems like a fortunate coincidence.
I think it'd be cleaner if we had separate methods for `nike.PublicKey` and `nike.PublicKey`, then we could also typecheck things (and avoid serialization to/from bytes)

Reply:
I don't understand what coincidence you are referring to. Private keys are NOT blinded. Private keys are merely a blinding factor.

Reply:
This is NOT an implementation detail. This code works for ALL NIKEs. All NIKEs MUST MUST perform the blinding operation with a blinding factor that is equal in length to the private key... and in fact it must be a valid private key for that NIKE scheme.